crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,384 @@
+        let rpo_falcon_proc = rpo_falcon_procs[0];
+
+        procedures.iter().for_each(|proc_info| {
+            if proc_info.mast_root() == &rpo_falcon_proc {
+                component_interface_set
+                    .insert(AccountComponentInterface::RpoFalcon512(proc_info.storage_offset()));
+            }
+        });
+
+        let mut custom_interface_procs_map: BTreeMap<u8, Vec<AccountProcedureInfo>> =
+            BTreeMap::new();
+        procedures.iter().for_each(|proc_info| {
+            // the meaning of this huge logical statement below is as follows:
+            // If we are examining a procedure from the basic wallet library, then it should be
+            // skipped, but only in case we already have basic wallet interface loaded. Motivation
+            // for that is that we should add procedures from the basic interfaces if they are
+            // included into some custom interface. Since the procedure duplication is not allowed,
+            // procedure should be included into the custom interface only if we haven't already
+            // loaded the corresponding basic interface. The same works for the procedures from the
+            // basic fungible faucet. RpoFalcon512 has only one procedure, so this statement could
+            // be simplified in that case.
+            if !(basic_wallet_library()
+                .mast_forest()
+                .procedure_digests()
+                .any(|wallet_proc_digest| &wallet_proc_digest == proc_info.mast_root())
+                && has_basic_wallet
+                || basic_fungible_faucet_library()
+                    .mast_forest()
+                    .procedure_digests()
+                    .any(|faucet_proc_digest| &faucet_proc_digest == proc_info.mast_root())
+                    && has_fungible_faucet)
+                && (&rpo_falcon_proc != proc_info.mast_root())
+            {
+                match custom_interface_procs_map.get_mut(&proc_info.storage_offset()) {
+                    Some(proc_vec) => proc_vec.push(*proc_info),
+                    None => {
+                        custom_interface_procs_map
+                            .insert(proc_info.storage_offset(), vec![*proc_info]);
+                    },
+                }
+            }
+        });
+
+        if !custom_interface_procs_map.is_empty() {
+            for proc_vec in custom_interface_procs_map.into_values() {
+                component_interface_set.insert(AccountComponentInterface::Custom(proc_vec));
+            }
+        }
+
+        component_interface_set
+    }
+
+    // probably it would be convenient to also have `from_components` constructor
+}
+
+// ACCOUNT COMPONENT INTERFACE
+// ================================================================================================
+
+/// Describes whether a note is compatible with a specific account.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum NoteAccountCompatibility {
```

Comment:
I agree with one of @PhilippGackstatter comments - let's get rid of this enum in favor of using simple booleans.

Reply:
Ah, yes - I don't think you've actually suggested using a boolean. I think I'm fine either way - with boolean and with `No/Maybe` enum (and maybe enum is a bit better).